cypress/integration/dashboard/dashboard_spec.js

```diff
@@ -0,0 +1,74 @@
+function archiveAllDashboards() {
+  cy.visit('/dashboards');
+
+  // archive all dashboards
+  cy.getByTestId('DashboardLayoutContent').then($wrapper => {
+    $wrapper.find('.table-main-title').each((_, { href }) => {
+      archiveDashboard(href);
+    });
+  });
+}
+
+function createNewDashboard(dashboardName) {
+  cy.visit('/dashboards');
+  cy.getByTestId('CreateButton').click();
+  cy.get('li[role="menuitem"]')
+    .contains('Dashboard')
+    .click();
+  cy.getByTestId('EditDashboardDialog').within(() => {
+    cy.getByTestId('DashboardSaveButton').should('be.disabled');
+    cy.get('input').type(dashboardName);
+    cy.getByTestId('DashboardSaveButton').click();
+  });
+}
+
+function archiveDashboard(url) {
+  if (url) {
+    cy.visit(url);
+  }
+
+  cy.getByTestId('DashboardMoreMenu')
+    .click()
+    .within(() => {
+      cy.get('li')
+        .contains('Archive')
+        .click();
+    });
+
+  cy.get('.btn-warning')
+    .contains('Archive')
+    .click();
+  cy.get('.label-tag-archived').should('exist');
+}
+
+describe('Dashboard', () => {
+  beforeEach(() => {
+    cy.login();
+    archiveAllDashboards();
+  });
+
+  afterEach(archiveAllDashboards);
```

Comment:
Cypress has some [good points](https://docs.cypress.io/guides/references/best-practices.html#Using-after-or-afterEach-hooks) not to use the `afterEach` to reset state. You actually solved half of that by also using the `beforeHook`, but I wonder if it's really worth it to have this here.

Reply:
Makes great sense 👍